src/System.ValueTuple/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -338,6 +338,7 @@ bool IStructuralEquatable.Equals(object other, IEqualityComparer comparer)
 
             var objTuple = (ValueTuple<T1>)other;
 
+            if (comparer == null) comparer = EqualityComparer<object>.Default;
             return comparer.Equals(Item1, objTuple.Item1);
```

Comment:
I wonder if there'd be any perf benefit to instead doing:
```C#
return comparer != null ?
    comparer.Equals(Item1, objTuple.Item1) :
    EqualityComparer<object>.Default.Equals(Item1, objTuple.Item1);
```
so that if it is null, we use virtual rather than interface dispatch.  But doing so also adds more code, and it's not clear to me that `null` is particularly common (after all it previously would have NRE'd), so may not even be worthwhile exploring.